ext/drsyms/libelftc/windows-mingw-port-common.patch

```diff
@@ -0,0 +1,1779 @@
+diff --git  common/_elftc.h common/_elftc.h
+index 5be8145b..898bee40 100644
+---  common/_elftc.h
++++ common/_elftc.h
+@@ -325,2 +325,6 @@ struct name {							\
+ 
++#ifndef ELFTC_VCSID
++#define ELFTC_VCSID(ID) /* nothing */
++#endif
++
+ #endif	/* ELFTC_VCSID */
+@@ -372,2 +376,10 @@ extern const char *__progname;
+ 
++#if defined _WIN32 && ! defined __CYGWIN__
++#include "windows-mingw/endian.h"
++
++#define	ELFTC_BYTE_ORDER			__BYTE_ORDER
++#define	ELFTC_BYTE_ORDER_LITTLE_ENDIAN		__LITTLE_ENDIAN
++#define	ELFTC_BYTE_ORDER_BIG_ENDIAN		__BIG_ENDIAN
++#endif
++
+ #if defined(__APPLE__)
+@@ -449,3 +461,7 @@ extern const char *__progname;
+ 
++#if !(defined _WIN32 && ! defined __CYGWIN__)
+ #include <sys/param.h>
++#else
++#include "windows-mingw/sys_param.h"
++#endif
+ #include <sys/endian.h>
+@@ -468,3 +484,7 @@ extern const char *__progname;
+ 
++#if !(defined _WIN32 && ! defined __CYGWIN__)
+ #include <sys/param.h>
++#else
++#include "windows-mingw/sys_param.h"
```

Comment:
There seem to be a ton of these `s|sys/|windows-mingw/sys_|`.  I would hope these could be used all at once with `-I` or `-isystem` and not need a ton of file name changes in the code: are they really named "sys_param.h" instead of "sys/param.h"?  That makes it really hard to interoperate.

Reply:
I added these headers to "common" directory.